Add traits and methods necessary for Architectural Map

nkinesis opened the pull request:
Problem: 
I got the following errors in the Architectural Map when generating it for FamixCPreproc entities:
- Instance of FamixCPreprocFolder did not understand #queryAllLocal
- Instance of FamixCPreprocInclude did not understand #isInvocation

Solution:
- Add the TEntityMetaLevelDependency trait to FamixCPreprocDiskEntity
- Remove the TEntityMetaLevelDependency trait FamixCPreprocCFile
- Implement the method isInvocation in FamixCPreprocInclude, returning false

NicolasAnquetil commented:
You did not do the change in the generator (and regenerate the model) ?
That would be better because somebody else could regenerate the meta-model and override your changes:

put `--|> #TEntityMetaLevelDependency.`  on `diskEntity` instead of `cFile`

For `#isInvocation` it is less clear because this metamodel does not define invocation ...
- may be this is another bug and creating `#isInvocation` does prevent it from appearing but does not really solves it
- could you split the solution in 2 so that we can accept the change for `#TEntityMetaLevelDependency` and think a bit about the invocation one ?
thanks

nkinesis commented:
Update:
- Removed #isInvocation and I will open a new PR to discuss this situation: https://github.com/moosetechnology/Famix-Cpp/pull/18
- Added #TEntityMetaLevelDependency to FamixCPreprocDiskEntity instead of FamixCPreprocCFile
